Sources/TuistGenerator/Generator/ProjectFileElements.swift

```diff
@@ -127,15 +127,32 @@ class ProjectFileElements {
         // Add the .gpx files if needed. GPS Exchange files must be added to the
         // project/workspace so that the scheme can correctly reference them.
         // In case the configuration already contains such file, we should avoid adding it twice
-        let gpxFiles = project.schemes.compactMap { scheme -> GroupFileElement? in
+        let runActionGPXFiles = project.schemes.compactMap { scheme -> GroupFileElement? in
             guard case let .gpxFile(path) = scheme.runAction?.options.simulatedLocation else {
                 return nil
             }
 
             return GroupFileElement(path: path, group: project.filesGroup)
         }
 
-        fileElements.formUnion(gpxFiles)
+        fileElements.formUnion(runActionGPXFiles)
+
+        let testActionGPXFiles = project.schemes.compactMap { scheme -> [GroupFileElement] in
+            guard let testAction = scheme.testAction else { return [] }
+
+            let elements = testAction.targets.compactMap { target -> GroupFileElement? in
+                guard case let .gpxFile(path) = target.simulatedLocation else {
+                    return nil
+                }
+
+                return GroupFileElement(path: path, group: project.filesGroup)
+            }
+
+            return elements
+        }
+        .flatMap { $0 }
+
+        fileElements.formUnion(testActionGPXFiles)
```

Comment:
Can we test the logic here in `ProjectFileElementsTests`?

Reply:
Added test cases for extracting GPX files.